instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/instrumentation.rb

```diff
@@ -20,7 +27,8 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         option :peer_service, default: nil, validate: :string
-        option :enable_sql_obfuscation, default: false, validate: :boolean
+        option :enable_sql_obfuscation, default: nil, validate: :boolean
```

Comment:
Does the default need to change here for the deprecation warning to work?

Reply:
that's an oopsie, I wanted to warn when enable_obfuscation is supplied. Let me fix it up.

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/894/commits/f02423a7799f7c18dd8ca9d2c2b6981ad4dc7bca

Reply:
Would it be unreasonable to add a `deprecated_option` helper to make this clearer to consumers (and, perhaps, to simplify option deprecation for instrumentation authors)? E.g.:
```suggestion
        deprecated_option :enable_sql_obfuscation, default: nil, validate: :boolean, alternative: :db_statement
```

Reply:
>Would it be unreasonable to add a deprecated_option helper to make this clearer to consumers (and, perhaps, to simplify option deprecation for instrumentation authors)? E.g.:

🤔 , I think adding something like this makes sense.  I'd want to see it come in as a follow up PR on its own where we can bikeshed the details of it.